### PR TITLE
fix (dave2d) : remove __NOP(); and  __BKPT(0);

### PR DESCRIPTION
**Pull request by jeremy-baker**
### Description of the feature or fix

fix (Dave2D) : Remove __NOP(); from the Dave2D code

Remove ARM __NOP(); from Dave2D code, to make compiling for RX core cleaner

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/draw/renesas/dave2d/lv_draw_dave2d_utils.c [hunk 1]
```diff
@@ -45,7 +45,7 @@ d2_color  lv_draw_dave2d_lv_colour_to_d2_colour(lv_color_t color)
     /*Color depth: 8 (A8), 16 (RGB565), 24 (RGB888), 32 (XRGB8888)*/
     switch(LV_COLOR_DEPTH) {
         case(8):
-            __BKPT(0);
+            LV_ASSERT(1 == 0);
```
Comment (C47D) [inline src/draw/renesas/dave2d/lv_draw_dave2d_utils.c]: Can this be reduced to `LV_ASSERT(0)`?

File: src/draw/renesas/dave2d/lv_draw_dave2d_utils.c [hunk 2]
```diff
@@ -113,7 +113,7 @@ d2_u32 lv_draw_dave2d_lv_colour_fmt_to_d2_fmt(lv_color_format_t colour_format)
             break;
 
         default:
-            __BKPT(0);
+            LV_ASSERT(1 == 0);
```
Comment (C47D) [inline src/draw/renesas/dave2d/lv_draw_dave2d_utils.c]: Can this be reduced to `LV_ASSERT(0)`?

File: src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c [hunk 3]
```diff
@@ -129,7 +127,7 @@ void lv_draw_dave2d_triangle(lv_draw_dave2d_unit_t * u, const lv_draw_triangle_d
         }
         else if(LV_GRAD_DIR_HOR == dsc->bg_grad.dir) {
             /* TODO */
-            __BKPT(0);
+            LV_ASSERT(1 == 0);
```
Comment (C47D) [inline src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c]: Can this be reduced to `LV_ASSERT(0)`?

File: src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c [hunk 4]
```diff
@@ -112,7 +110,7 @@ void lv_draw_dave2d_triangle(lv_draw_dave2d_unit_t * u, const lv_draw_triangle_d
 
             if(a1 < a2) {
                 /* TODO */
-                __BKPT(0);
+                LV_ASSERT(1 == 0);
```
Comment (C47D) [inline src/draw/renesas/dave2d/lv_draw_dave2d_triangle.c]: Can this be reduced to `LV_ASSERT(0)`?

**Review by kisvegabor — DISMISSED** [review]

**Review by C47D — APPROVED** [review]

File: src/draw/renesas/dave2d/lv_draw_dave2d_fill.c [hunk 5]
```diff
@@ -63,7 +61,7 @@ void lv_draw_dave2d_fill(lv_draw_dave2d_unit_t * u, const lv_draw_fill_dsc_t * d
 
             if(a1 < a2) {
                 /* TODO */
-                __BKPT(0);
+                LV_ASSERT(1 == 0);
```
Comment (XuNeo) [inline src/draw/renesas/dave2d/lv_draw_dave2d_fill.c]:
```suggestion
                LV_ASSERT(0);
```

File: src/draw/renesas/dave2d/lv_draw_dave2d_fill.c [hunk 6]
```diff
@@ -81,7 +79,7 @@ void lv_draw_dave2d_fill(lv_draw_dave2d_unit_t * u, const lv_draw_fill_dsc_t * d
         }
         else if(LV_GRAD_DIR_HOR == dsc->grad.dir) {
             /* TODO */
-            __BKPT(0);
+            LV_ASSERT(1 == 0);
```
Comment (XuNeo) [inline src/draw/renesas/dave2d/lv_draw_dave2d_fill.c]:
```suggestion
            LV_ASSERT(0);
```

File: src/draw/renesas/dave2d/lv_draw_dave2d_fill.c [hunk 7]
```diff
@@ -99,7 +97,7 @@ void lv_draw_dave2d_fill(lv_draw_dave2d_unit_t * u, const lv_draw_fill_dsc_t * d
 
             if(a1 < a2) {
                 /* TODO */
-                __BKPT(0);
+                LV_ASSERT(1 == 0);
```
Comment (XuNeo) [inline src/draw/renesas/dave2d/lv_draw_dave2d_fill.c]:
```suggestion
                LV_ASSERT(0);
```

**Review by XuNeo — APPROVED** [review]

**Review by kisvegabor — APPROVED** [review]

**Pull request merged**